File: category_service/app/dao/storage/storage.py
from abc import ABC, abstractmethod


class Storage(ABC):
    __slots__ = []

    @abstractmethod
    def find_one(self, *args, **kwargs):
        pass

    @abstractmethod
    def find(self, *args, **kwargs):
        pass

    @abstractmethod
    def create(self, *args, **kwargs):
        pass

    @abstractmethod
    def update(self, *args, **kwargs):
        pass

    @abstractmethod
    def delete(self, *args, **kwargs):
        pass

    @abstractmethod
    def execute(self, *args, **kwargs):
        pass

File: user_service/app/internal/user/storage.go
package user

import (
	"context"
)

type Storage interface {
	Create(ctx context.Context, user User) (string, error)
	FindByEmail(ctx context.Context, email string) (User, error)
	FindOne(ctx context.Context, uuid string) (User, error)
	Update(ctx context.Context, user User) error
	Delete(ctx context.Context, uuid string) error
}
